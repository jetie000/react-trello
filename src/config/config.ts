export const config = {
  API_URL: process.env.REACT_APP_API_URL,
  BURGER_BREAKPOINT: 576
}
