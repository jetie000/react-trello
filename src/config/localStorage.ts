export const localStorageConfig = {
  TOKEN_LOCALSTORAGE: "token",
  THEME_LOCALSTORAGE: "page_theme",
  LANGUAGE_LOCALSTORAGE: "page_language",
  USERID_LOCALSTORAGE: "user_id"
}
