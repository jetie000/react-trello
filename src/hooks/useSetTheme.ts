

export const useSetTheme = () => {
  const setTheme = () => {
    document.documentElement.setAttribute(
      "data-bs-theme",
      localStorage.getItem(localStorage.THEME_LOCALSTORAGE) || "dark"
    )
  }
  return setTheme
}
