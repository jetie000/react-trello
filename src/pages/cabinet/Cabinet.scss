@import "@/config/constants.scss";

.cabinet-my-info {
  width: 350px;
  flex-grow: 1;
}

@media (max-width: $media-sm-1) {
  .cabinet-my-info {
    width: 250px;
  }
  .cabinet-buttons {
    flex-direction: column;
  }
}
