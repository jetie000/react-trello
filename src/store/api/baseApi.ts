import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { variables } from "@/variables";
import { api } from "@/http";

const axiosBaseQuery =
  ({ baseUrl } = { baseUrl: '' }) =>
    async ({ url, method, data, params, headers }: any) => {
      try {
        const result = await api({
          url: baseUrl + url,
          method,
          data,
          params,
          headers,
        })
        return { data: result.data }
      } catch (axiosError: any) {
        const err = axiosError
        return {
          error: {
            status: err.response?.status,
            data: err.response?.data || err.message,
          },
        }
      }
    }

export const baseApi = createApi({
  reducerPath: 'baseApi',
  baseQuery: axiosBaseQuery({
    baseUrl: variables.API_URL,
  }),
  tagTypes: ['User', 'Users'],
  endpoints: () => ({}),
});