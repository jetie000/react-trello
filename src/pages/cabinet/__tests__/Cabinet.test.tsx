import * as reduxHooks from "react-redux"
import * as reactRouterDom from "react-router-dom"
import * as actions from "@/hooks/useActions"
import * as userApi from "@/store/api/user.api"
import { fireEvent, render, screen } from "@testing-library/react"
import Cabinet from "../Cabinet"
import { languages } from "@/config/languages"
import { IUser } from "@/types/user.interface"
import { mockReturnMutation } from "@/config/mockValues"

jest.mock("@/store/store")
jest.mock("react-redux")
jest.mock("react-router-dom")
jest.mock("@/hooks/useActions")
jest.mock("@/store/api/user.api")
const mockUseSelector = jest.spyOn(reduxHooks, "useSelector")
const mockUseActions = jest.spyOn(actions, "useActions")
const mockChange = jest.spyOn(userApi, "useChangeUserMutation")
const mockDelete = jest.spyOn(userApi, "useDeleteUserMutation")
const mockLogout = jest.spyOn(userApi, "useLogoutMutation")
const mockGet = jest.spyOn(userApi, "useGetByIdQuery")
const mockUseNavigate = jest.spyOn(reactRouterDom, "useNavigate")
const mockUseDispatch = jest.spyOn(reduxHooks, "useDispatch")
const mockReturnSelectorValue = {
  language: 1,
  id: 1,
  token: "token"
}
const mockNavigateReturn = jest.fn()
const mockDispatchReturn = jest.fn()
const mockChangeFunc = jest.fn()
const mockDeleteFunc = jest.fn()
const mockLogoutFunc = jest.fn()
const showToastMock = jest.fn()
const mockLogoutAction = jest.fn()
const mockReturnActionsValue = { showToast: showToastMock, logout: mockLogoutAction }

const nameUser = "John Doe"
const emailUser = "john.doe@example.com"
const oldPassUser = "password123"
const newPassUser = "password123456"

const mockReturnUser = (isLoading: boolean, isPressed: boolean) => ({
  isLoading: isLoading && isPressed,
  isError: false,
  isSuccess: !isLoading && isPressed,
  refetch: jest.fn(),
  data: {
    id: mockReturnSelectorValue.id,
    fullName: nameUser,
    access: true,
    email: emailUser
  } as IUser
})

describe("Cabinet", () => {
  beforeEach(() => {
    jest.restoreAllMocks()
    jest.resetAllMocks()
    mockDelete.mockReturnValue([mockDeleteFunc, mockReturnMutation(false, false)])
    mockLogout.mockReturnValue([mockLogoutFunc, mockReturnMutation(false, false)])
    mockGet.mockReturnValue(mockReturnUser(false, false))
    mockUseNavigate.mockReturnValue(mockNavigateReturn)
    mockUseSelector.mockReturnValue(mockReturnSelectorValue)
    mockUseDispatch.mockReturnValue(mockDispatchReturn)
    mockUseActions.mockReturnValue(mockReturnActionsValue as any)
    mockChange.mockReturnValue([mockChangeFunc, mockReturnMutation(false, false)])
  })
  it("should navigate to login page when token is not provided", () => {
    mockUseSelector.mockReturnValue({
      language: 1,
      id: undefined,
      token: undefined
    })
    render(<Cabinet />)
    expect(
      screen.queryByText(languages[mockReturnSelectorValue.language].MY_DATA)
    ).not.toBeInTheDocument()
  })
  it("should render component with all fields and buttons", () => {
    const component = render(<Cabinet />)
    expect(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_SURNAME_NAME)
    ).toBeInTheDocument()
    expect(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_NEW_EMAIL)
    ).toBeInTheDocument()
    expect(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_OLD_PASS)
    ).toBeInTheDocument()
    expect(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_NEW_PASSWORD)
    ).toBeInTheDocument()
    expect(
      screen.getByText(languages[mockReturnSelectorValue.language].MY_DATA)
    ).toBeInTheDocument()
    expect(
      screen.getByText(languages[mockReturnSelectorValue.language].CHANGE_DATA)
    ).toBeInTheDocument()
    expect(
      screen.getAllByText(languages[mockReturnSelectorValue.language].DELETE_ACCOUNT).length
    ).toEqual(2)
    expect(screen.getByText(languages[mockReturnSelectorValue.language].LOGOUT)).toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })
  it("should change user when all fields are filled correctly", () => {
    const component = render(<Cabinet />)
    expect(mockUseActions).toHaveBeenCalled()
    expect(mockChange).toHaveBeenCalled()
    expect(mockUseSelector).toHaveBeenCalled()
    fireEvent.change(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_SURNAME_NAME),
      {
        target: { value: nameUser }
      }
    )
    fireEvent.change(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_NEW_EMAIL),
      {
        target: { value: emailUser }
      }
    )
    fireEvent.change(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_OLD_PASS),
      {
        target: { value: oldPassUser }
      }
    )
    fireEvent.change(
      screen.getByPlaceholderText(languages[mockReturnSelectorValue.language].ENTER_NEW_PASSWORD),
      {
        target: { value: newPassUser }
      }
    )
    fireEvent.click(
      screen.getByRole("button", {
        name: languages[mockReturnSelectorValue.language].CHANGE_DATA
      })
    )
    mockChange.mockReturnValue([mockChangeFunc, mockReturnMutation(true, true)])
    component.rerender(<Cabinet />)
    mockChange.mockReturnValue([mockChangeFunc, mockReturnMutation(false, true)])
    component.rerender(<Cabinet />)
    expect(mockChangeFunc).toHaveBeenCalledWith({
      id: mockReturnSelectorValue.id,
      email: emailUser,
      oldPassword: oldPassUser,
      password: newPassUser,
      fullName: nameUser
    })
    expect(showToastMock).toHaveBeenCalledWith(
      languages[mockReturnSelectorValue.language].USER_SUCCESSFULLY_CHANGED
    )
    expect(showToastMock).toHaveBeenCalledTimes(1)
    expect(component).toMatchSnapshot()
  })
  it("should show enter data message when provided invalid data when changing user", () => {
    render(<Cabinet />)

    fireEvent.click(
      screen.getByRole("button", { name: languages[mockReturnSelectorValue.language].CHANGE_DATA })
    )
    expect(showToastMock).toHaveBeenCalledWith(
      languages[mockReturnSelectorValue.language].INPUT_DATA
    )
    expect(showToastMock).toHaveBeenCalledTimes(1)
  })

  it("should delete user", () => {
    const component = render(<Cabinet />)
    fireEvent.click(
      screen.getAllByText(languages[mockReturnSelectorValue.language].DELETE_ACCOUNT)[1]
    )
    expect(mockDeleteFunc).toHaveBeenCalledWith(mockReturnSelectorValue.id)
    expect(mockLogoutAction).toHaveBeenCalled()
  })

  it("should log out by clicking button", () => {
    const component = render(<Cabinet />)
    fireEvent.click(
      screen.getByRole("button", {
        name: languages[mockReturnSelectorValue.language].LOGOUT
      })
    )
    expect(mockLogoutFunc).toHaveBeenCalledWith(null)
    expect(mockNavigateReturn).toHaveBeenCalledWith("/login")
    expect(mockLogoutAction).toHaveBeenCalled()
    expect(mockDispatchReturn).toHaveBeenCalled()
  })
})
