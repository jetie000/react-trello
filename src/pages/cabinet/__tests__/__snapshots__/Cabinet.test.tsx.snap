// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cabinet should change user when all fields are filled correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" d-flex flex-fill flex-column"
      >
        <div
          class="ms-auto me-auto d-flex flex-column cabinet-my-info ps-3 pe-3 flex-fill"
        >
          <h2
            class="text-center p-3"
          >
            My info
          </h2>
          <div
            class="mb-3"
          >
            <label
              for="inputFullName"
            >
              Surname and name
            </label>
            <input
              class="form-control"
              id="inputFullName"
              placeholder="Enter surname and name"
              value="John Doe"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="inputEmail"
            >
              Email
            </label>
            <input
              class="form-control"
              id="inputEmail"
              placeholder="Input new email"
              type="email"
              value="john.doe@example.com"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="inputOldPassword"
            >
              Old password (necessary)
            </label>
            <input
              class="form-control"
              id="inputOldPassword"
              placeholder="Input old password"
              type="password"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="inputNewPassword"
            >
              New password (Optional)
            </label>
            <input
              class="form-control"
              id="inputNewPassword"
              placeholder="Input new password"
              type="password"
            />
          </div>
          <button
            class="btn btn-primary mt-3 w-100"
            type="button"
          >
            Change data
          </button>
        </div>
        <div
          class="d-flex justify-content-between cabinet-buttons mb-3"
        >
          <button
            class="justify-self-end btn btn-danger d-flex align-items-center gap-2 mt-4"
          >
            <svg
              class="bi bi-box-arrow-left"
              fill="currentColor"
              height="20"
              viewBox="0 0 16 16"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 12.5a.5.5 0 0 0 .5.5h8a.5.5 0 0 0 .5-.5v-9a.5.5 0 0 0-.5-.5h-8a.5.5 0 0 0-.5.5v2a.5.5 0 0 1-1 0v-2A1.5 1.5 0 0 1 6.5 2h8A1.5 1.5 0 0 1 16 3.5v9a1.5 1.5 0 0 1-1.5 1.5h-8A1.5 1.5 0 0 1 5 12.5v-2a.5.5 0 0 1 1 0z"
                fill-rule="evenodd"
              />
              <path
                d="M.146 8.354a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L1.707 7.5H10.5a.5.5 0 0 1 0 1H1.707l2.147 2.146a.5.5 0 0 1-.708.708z"
                fill-rule="evenodd"
              />
            </svg>
            Log out
          </button>
          <button
            class="justify-self-end btn btn-danger d-flex align-items-center gap-2 mt-4"
          >
            Delete an account
            <svg
              class="bi bi-trash3-fill"
              fill="currentColor"
              height="20"
              viewBox="0 0 16 16"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5Zm-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5ZM4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06Zm6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528ZM8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5Z"
              />
            </svg>
          </button>
        </div>
        <div
          aria-hidden="true"
          aria-labelledby="myInfoModalLabel"
          class="modal  fade"
          data-testid="myInfoModal"
          id="myInfoModal"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="modal-dialog modal-dialog-centered modal-sm"
            role="document"
          >
            <div
              class="modal-content"
            >
              <div
                class="modal-header"
              >
                <h5
                  class="modal-title"
                  id="myInfoModalLabel"
                >
                  Account deleting
                </h5>
                <button
                  aria-label="Close"
                  class="btn-close"
                  data-bs-dismiss="modal"
                  type="button"
                />
              </div>
              <div
                class="modal-body"
              >
                <div
                  class="d-flex flex-column gap-3"
                >
                  <span>
                    Are you sure you want to delete an account? You will also delete all of your collections.
                  </span>
                  <button
                    class="btn btn-danger"
                  >
                    Delete an account
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" d-flex flex-fill flex-column"
    >
      <div
        class="ms-auto me-auto d-flex flex-column cabinet-my-info ps-3 pe-3 flex-fill"
      >
        <h2
          class="text-center p-3"
        >
          My info
        </h2>
        <div
          class="mb-3"
        >
          <label
            for="inputFullName"
          >
            Surname and name
          </label>
          <input
            class="form-control"
            id="inputFullName"
            placeholder="Enter surname and name"
            value="John Doe"
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            for="inputEmail"
          >
            Email
          </label>
          <input
            class="form-control"
            id="inputEmail"
            placeholder="Input new email"
            type="email"
            value="john.doe@example.com"
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            for="inputOldPassword"
          >
            Old password (necessary)
          </label>
          <input
            class="form-control"
            id="inputOldPassword"
            placeholder="Input old password"
            type="password"
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            for="inputNewPassword"
          >
            New password (Optional)
          </label>
          <input
            class="form-control"
            id="inputNewPassword"
            placeholder="Input new password"
            type="password"
          />
        </div>
        <button
          class="btn btn-primary mt-3 w-100"
          type="button"
        >
          Change data
        </button>
      </div>
      <div
        class="d-flex justify-content-between cabinet-buttons mb-3"
      >
        <button
          class="justify-self-end btn btn-danger d-flex align-items-center gap-2 mt-4"
        >
          <svg
            class="bi bi-box-arrow-left"
            fill="currentColor"
            height="20"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 12.5a.5.5 0 0 0 .5.5h8a.5.5 0 0 0 .5-.5v-9a.5.5 0 0 0-.5-.5h-8a.5.5 0 0 0-.5.5v2a.5.5 0 0 1-1 0v-2A1.5 1.5 0 0 1 6.5 2h8A1.5 1.5 0 0 1 16 3.5v9a1.5 1.5 0 0 1-1.5 1.5h-8A1.5 1.5 0 0 1 5 12.5v-2a.5.5 0 0 1 1 0z"
              fill-rule="evenodd"
            />
            <path
              d="M.146 8.354a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L1.707 7.5H10.5a.5.5 0 0 1 0 1H1.707l2.147 2.146a.5.5 0 0 1-.708.708z"
              fill-rule="evenodd"
            />
          </svg>
          Log out
        </button>
        <button
          class="justify-self-end btn btn-danger d-flex align-items-center gap-2 mt-4"
        >
          Delete an account
          <svg
            class="bi bi-trash3-fill"
            fill="currentColor"
            height="20"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5Zm-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5ZM4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06Zm6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528ZM8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5Z"
            />
          </svg>
        </button>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="myInfoModalLabel"
        class="modal  fade"
        data-testid="myInfoModal"
        id="myInfoModal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered modal-sm"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <h5
                class="modal-title"
                id="myInfoModalLabel"
              >
                Account deleting
              </h5>
              <button
                aria-label="Close"
                class="btn-close"
                data-bs-dismiss="modal"
                type="button"
              />
            </div>
            <div
              class="modal-body"
            >
              <div
                class="d-flex flex-column gap-3"
              >
                <span>
                  Are you sure you want to delete an account? You will also delete all of your collections.
                </span>
                <button
                  class="btn btn-danger"
                >
                  Delete an account
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Cabinet should render component with all fields and buttons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" d-flex flex-fill flex-column"
      >
        <div
          class="ms-auto me-auto d-flex flex-column cabinet-my-info ps-3 pe-3 flex-fill"
        >
          <h2
            class="text-center p-3"
          >
            My info
          </h2>
          <div
            class="mb-3"
          >
            <label
              for="inputFullName"
            >
              Surname and name
            </label>
            <input
              class="form-control"
              id="inputFullName"
              placeholder="Enter surname and name"
              value="John Doe"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="inputEmail"
            >
              Email
            </label>
            <input
              class="form-control"
              id="inputEmail"
              placeholder="Input new email"
              type="email"
              value="john.doe@example.com"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="inputOldPassword"
            >
              Old password (necessary)
            </label>
            <input
              class="form-control"
              id="inputOldPassword"
              placeholder="Input old password"
              type="password"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="inputNewPassword"
            >
              New password (Optional)
            </label>
            <input
              class="form-control"
              id="inputNewPassword"
              placeholder="Input new password"
              type="password"
            />
          </div>
          <button
            class="btn btn-primary mt-3 w-100"
            type="button"
          >
            Change data
          </button>
        </div>
        <div
          class="d-flex justify-content-between cabinet-buttons mb-3"
        >
          <button
            class="justify-self-end btn btn-danger d-flex align-items-center gap-2 mt-4"
          >
            <svg
              class="bi bi-box-arrow-left"
              fill="currentColor"
              height="20"
              viewBox="0 0 16 16"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 12.5a.5.5 0 0 0 .5.5h8a.5.5 0 0 0 .5-.5v-9a.5.5 0 0 0-.5-.5h-8a.5.5 0 0 0-.5.5v2a.5.5 0 0 1-1 0v-2A1.5 1.5 0 0 1 6.5 2h8A1.5 1.5 0 0 1 16 3.5v9a1.5 1.5 0 0 1-1.5 1.5h-8A1.5 1.5 0 0 1 5 12.5v-2a.5.5 0 0 1 1 0z"
                fill-rule="evenodd"
              />
              <path
                d="M.146 8.354a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L1.707 7.5H10.5a.5.5 0 0 1 0 1H1.707l2.147 2.146a.5.5 0 0 1-.708.708z"
                fill-rule="evenodd"
              />
            </svg>
            Log out
          </button>
          <button
            class="justify-self-end btn btn-danger d-flex align-items-center gap-2 mt-4"
          >
            Delete an account
            <svg
              class="bi bi-trash3-fill"
              fill="currentColor"
              height="20"
              viewBox="0 0 16 16"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5Zm-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5ZM4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06Zm6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528ZM8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5Z"
              />
            </svg>
          </button>
        </div>
        <div
          aria-hidden="true"
          aria-labelledby="myInfoModalLabel"
          class="modal  fade"
          data-testid="myInfoModal"
          id="myInfoModal"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="modal-dialog modal-dialog-centered modal-sm"
            role="document"
          >
            <div
              class="modal-content"
            >
              <div
                class="modal-header"
              >
                <h5
                  class="modal-title"
                  id="myInfoModalLabel"
                >
                  Account deleting
                </h5>
                <button
                  aria-label="Close"
                  class="btn-close"
                  data-bs-dismiss="modal"
                  type="button"
                />
              </div>
              <div
                class="modal-body"
              >
                <div
                  class="d-flex flex-column gap-3"
                >
                  <span>
                    Are you sure you want to delete an account? You will also delete all of your collections.
                  </span>
                  <button
                    class="btn btn-danger"
                  >
                    Delete an account
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" d-flex flex-fill flex-column"
    >
      <div
        class="ms-auto me-auto d-flex flex-column cabinet-my-info ps-3 pe-3 flex-fill"
      >
        <h2
          class="text-center p-3"
        >
          My info
        </h2>
        <div
          class="mb-3"
        >
          <label
            for="inputFullName"
          >
            Surname and name
          </label>
          <input
            class="form-control"
            id="inputFullName"
            placeholder="Enter surname and name"
            value="John Doe"
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            for="inputEmail"
          >
            Email
          </label>
          <input
            class="form-control"
            id="inputEmail"
            placeholder="Input new email"
            type="email"
            value="john.doe@example.com"
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            for="inputOldPassword"
          >
            Old password (necessary)
          </label>
          <input
            class="form-control"
            id="inputOldPassword"
            placeholder="Input old password"
            type="password"
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            for="inputNewPassword"
          >
            New password (Optional)
          </label>
          <input
            class="form-control"
            id="inputNewPassword"
            placeholder="Input new password"
            type="password"
          />
        </div>
        <button
          class="btn btn-primary mt-3 w-100"
          type="button"
        >
          Change data
        </button>
      </div>
      <div
        class="d-flex justify-content-between cabinet-buttons mb-3"
      >
        <button
          class="justify-self-end btn btn-danger d-flex align-items-center gap-2 mt-4"
        >
          <svg
            class="bi bi-box-arrow-left"
            fill="currentColor"
            height="20"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 12.5a.5.5 0 0 0 .5.5h8a.5.5 0 0 0 .5-.5v-9a.5.5 0 0 0-.5-.5h-8a.5.5 0 0 0-.5.5v2a.5.5 0 0 1-1 0v-2A1.5 1.5 0 0 1 6.5 2h8A1.5 1.5 0 0 1 16 3.5v9a1.5 1.5 0 0 1-1.5 1.5h-8A1.5 1.5 0 0 1 5 12.5v-2a.5.5 0 0 1 1 0z"
              fill-rule="evenodd"
            />
            <path
              d="M.146 8.354a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L1.707 7.5H10.5a.5.5 0 0 1 0 1H1.707l2.147 2.146a.5.5 0 0 1-.708.708z"
              fill-rule="evenodd"
            />
          </svg>
          Log out
        </button>
        <button
          class="justify-self-end btn btn-danger d-flex align-items-center gap-2 mt-4"
        >
          Delete an account
          <svg
            class="bi bi-trash3-fill"
            fill="currentColor"
            height="20"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5Zm-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5ZM4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06Zm6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528ZM8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5Z"
            />
          </svg>
        </button>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="myInfoModalLabel"
        class="modal  fade"
        data-testid="myInfoModal"
        id="myInfoModal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered modal-sm"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            >
              <h5
                class="modal-title"
                id="myInfoModalLabel"
              >
                Account deleting
              </h5>
              <button
                aria-label="Close"
                class="btn-close"
                data-bs-dismiss="modal"
                type="button"
              />
            </div>
            <div
              class="modal-body"
            >
              <div
                class="d-flex flex-column gap-3"
              >
                <span>
                  Are you sure you want to delete an account? You will also delete all of your collections.
                </span>
                <button
                  class="btn btn-danger"
                >
                  Delete an account
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
